php-frontend/src/main/java/org/sonar/php/parser/NewPHPGrammar.java

```diff
@@ -0,0 +1,1479 @@
+          b.token(PHPLexicalGrammar.CLASS_CONSTANT),
+          b.token(PHPLexicalGrammar.FILE_CONSTANT),
+          b.token(PHPLexicalGrammar.DIR_CONSTANT),
+          b.token(PHPLexicalGrammar.FUNCTION_CONSTANT),
+          b.token(PHPLexicalGrammar.LINE_CONSTANT),
+          b.token(PHPLexicalGrammar.METHOD_CONSTANT),
+          b.token(PHPLexicalGrammar.NAMESPACE_CONSTANT),
+          b.token(PHPLexicalGrammar.TRAIT_CONSTANT)))));
+  }
+
+  public LiteralTree NUMERIC_LITERAL() {
+    return b.<LiteralTree>nonterminal(Kind.NUMERIC_LITERAL).is(
+      f.numericLiteral(b.token(PHPLexicalGrammar.NUMERIC_LITERAL)));
+  }
+
+  public ExpressionTree STRING_LITERAL() {
+    return b.<ExpressionTree>nonterminal().is(
+      b.firstOf(
+        f.regularStringLiteral(b.token(PHPLexicalGrammar.REGULAR_STRING_LITERAL)),
+        EXPANDABLE_STRING_LITERAL()));
+  }
+
+  public ExpandableStringLiteralTree EXPANDABLE_STRING_LITERAL() {
+    return b.<ExpandableStringLiteralTree>nonterminal(Kind.EXPANDABLE_STRING_LITERAL).is(
+      f.expandableStringLiteral(
+        b.token(PHPLexicalGrammar.SPACING),
+        b.token(PHPLexicalGrammar.DOUBLE_QUOTE),
+        b.oneOrMore(
+          b.firstOf(
+            ENCAPSULATED_STRING_VARIABLE(),
+            EXPANDABLE_STRING_CHARACTERS())),
+        b.token(PHPLexicalGrammar.DOUBLE_QUOTE)));
+  }
+
+  public ExpressionTree ENCAPSULATED_STRING_VARIABLE() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.ENCAPS_VAR).is(
+      b.firstOf(
+        ENCAPSULATED_SEMI_COMPLEX_VARIABLE(),
+        ENCAPSULATED_SIMPLE_VARIABLE(),
+        ENCAPSULATED_COMPLEX_VARIABLE()));
+  }
+
+  public ExpressionTree ENCAPSULATED_COMPLEX_VARIABLE() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.COMPLEX_ENCAPS_VARIABLE).is(
+      f.encapsulatedComplexVariable(
+        b.token(LCURLYBRACE),
+        b.token(PHPLexicalGrammar.NEXT_IS_DOLLAR),
+        EXPRESSION(),
+        b.token(RCURLYBRACE)));
+  }
+
+  public ExpressionTree ENCAPSULATED_SEMI_COMPLEX_VARIABLE() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.SEMI_COMPLEX_ENCAPS_VARIABLE).is(
+      f.encapsulatedSemiComplexVariable(
+        b.token(DOLLAR_LCURLY),
+        b.firstOf(
+          EXPRESSION(),
+          f.expressionRecovery(b.token(PHPLexicalGrammar.SEMI_COMPLEX_RECOVERY_EXPRESSION))),
+        b.token(RCURLYBRACE)));
+  }
+
+  public ExpressionTree ENCAPSULATED_SIMPLE_VARIABLE() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.SIMPLE_ENCAPS_VARIABLE).is(
+      f.encapsulatedSimpleVar(ENCAPSULATED_VARIABLE_IDENTIFIER(),
+        b.optional(b.firstOf(
+          f.expandableArrayAccess(
+            b.token(LBRACKET),
+            b.firstOf(IDENTIFIER(), NUMERIC_LITERAL(), ENCAPSULATED_VARIABLE_IDENTIFIER()),
+            b.token(RBRACKET)),
+          f.expandableObjectMemberAccess(
+            b.token(ARROW),
+            IDENTIFIER())))));
+  }
+
+  public IdentifierTree IDENTIFIER() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00100)